types/src/genesis/mod.rs

```diff
@@ -2,47 +2,55 @@
 // SPDX-License-Identifier: Apache-2.0, MIT
 
 use super::SectorSize;
-use address::Address;
-use num_bigint::bigint_ser;
-use serde::Serialize;
+use address::{json as addr_json, Address};
+use chrono::Utc;
+use num_bigint::bigint_ser::json as bigint_json;
+use serde::{Deserialize, Serialize};
 use vm::TokenAmount;
 
-#[derive(Serialize)]
+#[derive(Serialize, Deserialize)]
 #[serde(rename_all = "lowercase")]
 pub enum ActorType {
     Account,
     MultiSig,
 }
 
-#[derive(Serialize)]
+#[derive(Serialize, Deserialize)]
 #[serde(rename_all = "PascalCase")]
 pub struct Actor {
     pub actor_type: ActorType,
-    #[serde(with = "bigint_ser")]
-    pub token_amount: TokenAmount,
+    #[serde(with = "bigint_json")]
+    pub balance: TokenAmount,
+
+    #[serde(with = "addr_json")]
+    pub owner: Address,
 }
 
-#[derive(Serialize)]
+#[derive(Serialize, Deserialize)]
 #[serde(rename_all = "PascalCase")]
 pub struct Miner {
+    #[serde(with = "addr_json")]
     pub owner: Address,
+
+    #[serde(with = "addr_json")]
     pub worker: Address,
-    pub peer_id: Vec<u8>,
+    pub peer_id: String,
 
-    #[serde(with = "bigint_ser")]
+    #[serde(with = "bigint_json")]
     pub market_balance: TokenAmount,
-    #[serde(with = "bigint_ser")]
+    #[serde(with = "bigint_json")]
     pub power_balance: TokenAmount,
     pub sector_size: SectorSize,
+    // pub sectors: Vec<PreSeal>,
```

Comment:
intentionally commented out?

Reply:
Yeah, should've left a comment. I didn't find PreSeal struct anywhere and thought it's not implemented yet? :thinking: Otherwise I can add it as well.

Reply:
Nah, leave it, no problem. Can come later when needed

Reply:
removed